src/handlers/uniswap.ts

```diff
@@ -31,14 +35,121 @@ import {
   subtract,
 } from '@rainbow-me/helpers/utilities';
 import { Network } from '@rainbow-me/networkTypes';
-import { ethUnits, UNISWAP_TESTNET_TOKEN_LIST } from '@rainbow-me/references';
+import {
+  erc20ABI,
+  ethUnits,
+  UNISWAP_TESTNET_TOKEN_LIST,
+} from '@rainbow-me/references';
 import { ethereumUtils, logger } from '@rainbow-me/utils';
 
 export enum Field {
   INPUT = 'INPUT',
   OUTPUT = 'OUTPUT',
 }
 
+const MAX_GAS_LIMIT = 460000;
+const GAS_LIMIT_INCREMENT = 50000;
+const EXTRA_GAS_PADDING = 1.5;
+
+export const getStateDiff = async (
+  provider: StaticJsonRpcProvider,
+  tradeDetails: Quote
+): Promise<any> => {
+  const tokenAddress = tradeDetails.sellTokenAddress;
+  const fromAddr = tradeDetails.from;
+  const toAddr = RAINBOW_ROUTER_CONTRACT_ADDRESS;
+  const tokenContract = new ethers.Contract(tokenAddress, erc20ABI, provider);
+
+  // Get data
+  const { data } = await tokenContract.populateTransaction.approve(
+    toAddr,
+    ethers.constants.MaxUint256.toHexString()
+  );
+
+  // trace_call default params
+  const callParams = [
+    {
+      data,
+      from: fromAddr,
+      to: tokenAddress,
+      value: '0x0',
+    },
+    ['stateDiff'],
+    'latest',
+  ];
+
+  const trace = await provider.send('trace_call', callParams);
+
+  if (trace.stateDiff) {
+    const slotAddress = Object.keys(trace.stateDiff[tokenAddress].storage)[0];
```

Comment:
i'm getting an error message when calculating gas limit with fake approval 
`Blew up trying to get state diff. Falling back to defaults [TypeError: Cannot read property 'storage' of undefined]`
i get the gas limit will go to the default value, so nothing is really failing

Reply:
hm actually the fee shown now is 25% higher than the default since now we get the default 300000

Reply:
just changed the code to not blow up by using optionals.